js/main.js

```diff
@@ -150,19 +186,102 @@ function showMoviePage() {
   var $searchResult = document.querySelector('[data-view="search-result"]');
   $moviePage.classList.remove('hidden');
   $navForm.classList.add('hidden');
-  $searchResult.classList.add('hidden');
+  if ($searchResult) {
+    $searchResult.classList.add('hidden');
+  }
 }
 
 $plus.addEventListener('click', function () {
-  data.list.array.unshift(data.viewing.currentlyViewing);
+  data.list.array.unshift(data.movieView.currentlyViewing);
   checkList();
 });
 
 function checkList() {
+  var includes = false;
   for (var i = 0; i < data.list.array.length; i++) {
-    if (data.viewing.currentlyViewing.imdbID === data.list.array[i].imdbID) {
-      $plus.classList.add('hidden');
-      $check.classList.remove('hidden');
+    if (data.movieView.currentlyViewing.imdbID === data.list.array[i].imdbID) {
+      includes = true;
     }
   }
+
+  if (includes === true) {
+    $plus.classList.add('hidden');
+    $check.classList.remove('hidden');
+  } else {
+    $plus.classList.remove('hidden');
+    $check.classList.add('hidden');
+  }
+
+}
+
+function viewList() {
+  var $searchResult = document.querySelector('[data-view="search-result"]');
+  if (data.list.viewing === true) {
+    closeList();
+  }
+  $list.classList.add('hidden');
+  $moviePage.classList.add('hidden');
+  data.list.viewing = true;
+  $navForm.classList.add('hidden');
+  $homePage.classList.add('hidden');
+  $nav.classList.remove('hidden');
+  if ($searchResult) {
+    $searchResult.classList.add('hidden');
+  }
+  $main.appendChild(createList());
+}
+
+function createList() {
+  var tempList = data.list.array;
+  var container = document.createElement('div');
+  container.setAttribute('class', 'container text-center');
+  container.setAttribute('data-view', 'list-page');
+
+  var row = document.createElement('div');
+  row.setAttribute('class', 'row');
+
+  var columnOneFourth = document.createElement('div');
+  columnOneFourth.className = 'column-one-fourth list-close-container';
+  row.appendChild(columnOneFourth);
+
+  var close = document.createElement('i');
+  close.className = 'fa-solid fa-xmark';
+  close.setAttribute('id', 'list-close');
+  columnOneFourth.appendChild(close);
+
+  var row2 = document.createElement('div');
+  row2.className = 'row';
+
+  for (var i = 0; i < tempList.length; i++) {
+    row2.appendChild(createColumn(i));
+  }
+
+  function createColumn(index) {
+    var column = document.createElement('div');
+    column.setAttribute('class', 'column-one-third margin-auto');
+    var img = document.createElement('img');
+    img.setAttribute('src', tempList[index].Poster);
+    column.appendChild(img);
+    return column;
+  }
```

Comment:
Instead of just giving the index as an argument, you should give the value at that index as an argument, and that will allow you to move this definition outside of the function definition it's inside of.